lib/glific/partners.ex

```diff
@@ -354,6 +355,23 @@ defmodule Glific.Partners do
     end
   end
 
+  @doc """
+  Returns quality rating information for an organization provider
+  """
+  @spec get_quality_rating(pos_integer()) :: {:ok, any()} | {:error, String.t()}
+  def get_quality_rating(organization_id) do
+    organization = organization(organization_id)
+
+    if is_nil(organization.services["bsp"]) do
```

Comment:
Do we really need this check? 
I think we can assume that if no BSP is active then we don't fetch this service at all.

Reply:
Anyway the API will be open to use to adding a check makes it more solid and returns proper error. Let me know if it needs to be removed still